integration/disk_cleanup_helpers.go

```diff
@@ -0,0 +1,171 @@
+// +build integration
+
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3db/storage"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/ts"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	errDataCleanupTimedOut = errors.New("cleaning up data files took too long")
+)
+
+func createWriter(storageOpts storage.Options) fs.FileSetWriter {
```

Comment:
nit: For consistency for constructing things perhaps you can call this `newWriter(...)`?